src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -514,6 +514,41 @@ private int accessTokenExpirationBufferTime
 
         #endregion
 
+        #region Public and Internal Methods
+
+        // @TODO: Make internal by making the SqlInternalConnection implementation internal
+        public override void Dispose()
+        {
+            SqlClientEventSource.Log.TryAdvancedTraceEvent(
+                $"SqlInternalConnectionTds.Dispose | ADV | " +
+                $"Object ID {ObjectID} disposing");
+
+            try
+            {
+                // Guard against multiple concurrent dispose calls -- Delegated Transactions might
+                // cause this.
+                TdsParser parser = Interlocked.Exchange(ref _parser, null);
+
+                Debug.Assert(parser is not null && _fConnectionOpen || parser is null && !_fConnectionOpen,
+                    "Unexpected state on dispose");
+
+                parser?.Disconnect();
```

Comment:
Rewritten to use conditional access (rather than `if (x is not null) { x.disconnect }`)